src/Microsoft.VisualStudio.Threading.Analyzers.Tests/MultiAnalyzerTests.cs

```diff
@@ -266,37 +284,43 @@ public void NoValueTupleReference()
         }
 
         [Fact]
-        public void NameOfUsedInAttributeArgument()
+        public async Task NameOfUsedInAttributeArgument()
         {
             var test = @"
 [System.Diagnostics.DebuggerDisplay(""hi"", Name = nameof(System.Console))]
 class Foo { }
 ";
-            this.VerifyCSharpDiagnostic(test);
+            await Verify.VerifyAnalyzerAsync(test);
         }
 
-        private void VerifyNoMoreThanOneDiagnosticPerLine(string test, bool hasEntrypoint = false)
+        public static class Verifier
         {
-            this.LogFileContent(test);
-            ImmutableArray<DiagnosticAnalyzer> analyzers = this.GetCSharpDiagnosticAnalyzers();
-            var actualResults = GetSortedDiagnostics(new[] { test }, LanguageNames.CSharp, analyzers, hasEntrypoint);
-            string diagnosticsOutput = actualResults.Any() ? FormatDiagnostics(analyzers, actualResults.ToArray()) : "    NONE.";
-            this.logger.WriteLine("Actual diagnostics:\n" + diagnosticsOutput);
-
-            // Assert that each line only merits at most one diagnostic.
-            int lastDiagnosticLine = -1;
-            Diagnostic lastDiagnostic = null;
-            for (int i = 0; i < actualResults.Length; i++)
+            public static DiagnosticResult Diagnostic(DiagnosticDescriptor descriptor)
+                => new DiagnosticResult(descriptor);
+
+            public static DiagnosticResult CompilerError(string errorIdentifier)
+                => new DiagnosticResult(errorIdentifier, DiagnosticSeverity.Error);
+
+            public static Task VerifyAnalyzerAsync(string source, params DiagnosticResult[] expected)
             {
-                Diagnostic diagnostic = actualResults[i];
-                int diagnosticLinePosition = diagnostic.Location.GetLineSpan().StartLinePosition.Line;
-                if (lastDiagnosticLine == diagnosticLinePosition)
+                var test = new Test
                 {
-                    Assert.False(true, $"Both {lastDiagnostic.Id} and {diagnostic.Id} produced diagnostics for line {diagnosticLinePosition + 1}.");
-                }
+                    TestCode = source,
+                };
 
-                lastDiagnosticLine = diagnosticLinePosition;
-                lastDiagnostic = diagnostic;
+                test.ExpectedDiagnostics.AddRange(expected);
+                return test.RunAsync();
+            }
+
+            public class Test : CSharpAnalyzerTest<VSTHRD002UseJtfRunAnalyzer>
```

Comment:
💭 @jmarolf This is a bit weird. I had to pick one of the analyzers to include in the base definition, and then I overrode `GetDiagnosticAnalyzers` to return many more.